hs_core/hydroshare/resource.py

```diff
@@ -998,6 +999,189 @@ def get_crossref_url():
     return main_url
 
 
+def get_datacite_url():
+    main_url = settings.TEST_DATACITE_API_URL
+    if not settings.USE_DATACITE_TEST:
+        main_url = settings.DATACITE_API_URL
+    return main_url
+
+
+def deposit_res_metadata_with_datacite(res):
+    """
+    Deposit resource metadata with DataCite using the Fabrica-style payload.
+    Args:
+        res: Django model instance with metadata
+
+    Returns:
+        Response object or None if error occurred
+    """
+
+    try:
+        token = base64.b64encode(f"{settings.DATACITE_USERNAME}:{settings.DATACITE_PASSWORD}".encode()).decode()
```

Comment:
`DATACITE_USERNAME` and `DATACITE_PASSWORD` are not in your settings.py nor in local_settings.py

Reply:
If you're not okay having the sandbox creds made public, you will need to add them as GH environment vars and then figure out how to get them into Django within the GH action

Reply:
https://github.com/hydroshare/hydroshare/settings/secrets/actions